### PR TITLE
cisco-meraki

### PR DESCRIPTION
**Pull request by asafshen**
fixes https://github.com/demisto/server/issues/6334

implementation of most API calls

### CODE REVIEW

File: Integrations/integration-cisco-meraki.yml [hunk 1]
```diff
@@ -14,10 +14,13 @@ configuration:
...
-
-    var merakiUrlPrefix = 'https://dashboard.meraki.com/api/v0/';
-
+  script: |
+    /**
+       * flatten nested object to top level fields
+       * @param {Object | Array<Object>} ob - object to flatten
+       * @returns {String} the flatten object
+       * { "a": "bla", "b": { "c": "hello", "d": "there" } } => { "a": "bla", "b.c" = "hello", "b.d": "there" }
+    */
```
Comment (Spivaka) [inline Integrations/integration-cisco-meraki.yml]: should be in common server for a generic usage

File: Integrations/integration-cisco-meraki.yml [hunk 2]
```diff
@@ -38,12 +41,10 @@ script:
         return toReturn;
     };
 
-    // ------------------------ mics -------------------- //
-
     /**
        * Converts a demisto table in JSON form to a Markdown table
        * @param {String} name - the name of the table
-       * @param {Object} table - the JSON table - Array of objects with the same keys
+       * @param {Object | Array<Object>} table - the JSON table - Array of objects with the same keys
```
Comment (Spivaka) [inline Integrations/integration-cisco-meraki.yml]: Also look like a function for general use (and a well written one, :+1:) so please move to common

File: Integrations/integration-cisco-meraki.yml [hunk 3]
```diff
@@ -102,28 +103,39 @@ script:
...
+    var apiKey = params.apiKey;
+
     var sendRequest = function(url, method, body) {
         var requestUrl = merakiUrlPrefix + url;
-        var apiKey = params.apiKey;
         var requestMethod = method || 'GET';
 
-
         var httpParams = {
             Method: requestMethod,
```
Comment (Spivaka) [inline Integrations/integration-cisco-meraki.yml]: kine 111 should be here, no other usage of method. probably the same for line 110

File: Integrations/integration-cisco-meraki.yml [hunk 4]
```diff
@@ -102,28 +103,39 @@ script:
...
             Headers: {
                 'X-Cisco-Meraki-API-Key': [apiKey],
                 'Content-Type': ['application/json'],
             },
+            body: body
```
Comment (Spivaka) [inline Integrations/integration-cisco-meraki.yml]: add insecure and proxy here

Comment (Spivaka) [inline Integrations/integration-cisco-meraki.yml]: I think body should be Body in order to be marshalled

File: Integrations/integration-cisco-meraki.yml [hunk 5]
```diff
@@ -102,28 +103,39 @@ script:
...
         var res = http(
             requestUrl,
-            httpParams
+            httpParams,
+            false,
+            false,
+            true // no redirect
         );
 
+        if (res.StatusCode === 302) {
```
Comment (Spivaka) [inline Integrations/integration-cisco-meraki.yml]: How can we be redirected if the no redirect is on by default?

File: Integrations/integration-cisco-meraki.yml [hunk 6]
```diff
@@ -135,7 +147,7 @@ script:
             // body is empty - than just return true (action succeed)
```
Comment (Spivaka) [inline Integrations/integration-cisco-meraki.yml]: condition doesn't match comment... fix one of them (probably the code as the comment makes sense)

File: Integrations/integration-cisco-meraki.yml [hunk 7]
```diff
@@ -135,7 +147,7 @@ script:
...
             return true;
         } else {
-             var body = res && JSON.parse(res.Body);
+             var body = JSON.parse(res.Body);
```
Comment (Spivaka) [inline Integrations/integration-cisco-meraki.yml]: I think the older version was better, would also add res && res.Body

File: Integrations/integration-cisco-meraki.yml (hunk 5 above)
Comment (asafshen) [inline Integrations/integration-cisco-meraki.yml]: the 'no-redirect' means that http-client is not automatically redirects the request - instead it returns 302/3/4

maybe we should change its name if its not clear. anyway I will add better doc on this integration

**Pull request merged**